src/Extension.Tests/ChallengeTests.cs

```diff
@@ -66,10 +58,7 @@ public async Task teacher_can_access_code_from_submission_history_when_evaluatin
         public async Task challenge_tracks_submitted_code_in_submission_history()
         {
             var lesson = new Lesson();
-            using var kernel = new CompositeKernel
-            {
-                new CSharpKernel()
-            }.UseLessonEvaluateMiddleware(lesson);
+            using var kernel = CreateKernel(lesson);
             var challenge = GetEmptyChallenge();
             await lesson.StartChallengeAsync(challenge);
```

Comment:
same here, starting a challenge that is not part of a lesson should not be allowed